noah/src/main/scala/edu/uci/ics/cloudberry/noah/kafka/AsterixConsumerKafka.scala

```diff
@@ -45,53 +44,42 @@ class AsterixConsumerKafka(config: Config) {
     return props
   }
 
-  private def ingest(records: ConsumerRecords[String, String]) {
-    val feedDriver: TwitterFeedStreamDriver = new TwitterFeedStreamDriver
-    val socketAdapterClient = feedDriver.openSocket(config)
-      try {
+  def subscribe(consumer: KafkaConsumer[String, String], source: Source): Unit = {
+    val topics: Array[String] = Array(config.getTopic(source))
+    consumer.subscribe(topics.toSeq)
+  }
+
+  def sendToAsterix(source: Source, url: String, dataset: String, asterixDataInsertion: AsterixDataInsertion, wsClient: AhcWSClient, records: ConsumerRecords[String, String]): Unit = {
+    if (!records.isEmpty) {
+      if (source == Config.Source.Zika) {
+        asterixDataInsertion.ingest(records, config)
+      }
+      else {
         for (record <- records) {
-          val adm: String = TagTweet.tagOneTweet(record.value, false)
-          socketAdapterClient.ingest(adm)
+          asterixDataInsertion.insertRecord(url, config.getDataverse, dataset, record.value, wsClient)
         }
       }
-      catch {
-        case e: TwitterException => {
-          e.printStackTrace(System.err)
-      }
-      case e: CmdLineException => {
-        e.printStackTrace(System.err)
-      }
-      case e: Exception => {
-        e.printStackTrace(System.err)
-      }
-    } finally {
-        if (socketAdapterClient != null) {
-          socketAdapterClient.finalize
-      }
     }
   }
 
   @throws[CmdLineException]
-  def consume(source: Source) {
-    val props = getProperties
-    val consumer = new KafkaConsumer[String, String](props)
-    val topics: Array[String] = Array(config.getTopic(source))
-    consumer.subscribe(topics.toSeq)
+  def consume(source: Source, consumer: KafkaConsumer[String, String], timeout: Long) {
+    subscribe(consumer, source)
+    val asterixDataInsertion = new AsterixDataInsertion()
+    val wsClient = asterixDataInsertion.createClient()
     val dataset = config.getDataset(source)
     val url = s"${config.getAxServer}/aql"
-    val wsClient: AhcWSClient = AsterixHttpRequest.createClient()
-    while (true) {
-      val records: ConsumerRecords[String, String] = consumer.poll(props.getProperty("poll.ms").toLong)
-
-      if (source == Config.Source.Zika) {
-        ingest(records)
+    try {
+      while (true) {
```

Comment:
do you want to add the sleep logic? 
you can add it later, in which case it's better to add a `//TODO xxx` comment.